oak_functions/client/java/src/com/google/oak/functions/client/AttestationClient.java

```diff
@@ -123,21 +123,25 @@ public void onCompleted() {}
         new ClientAttestationEngine(TEST_TEE_MEASUREMENT.getBytes(UTF_8));
 
     // Send client hello message.
-    ClientHello clientHello = attestationEngine.createClientHello();
+    byte[] clientHello = attestationEngine.createClientHello();
     AttestedInvokeRequest clientHelloRequest =
-        AttestedInvokeRequest.newBuilder().setClientHello(clientHello).build();
+        AttestedInvokeRequest.newBuilder().setAttestationMessage(
+            AttestationMessage.newBuilder().setBody(ByteString.copyFrom(clientHello)).build()
+        ).build();
     requestObserver.onNext(clientHelloRequest);
 
     // Receive server attestation identity containing server's ephemeral public key.
     AttestedInvokeResponse serverIdentityResponse = messageQueue.take();
-    ServerIdentity serverIdentity = serverIdentityResponse.getServerIdentity();
+    byte[] serverIdentity = serverIdentityResponse.getAttestationMessage().getBody().toByteArray();
 
     // Remotely attest the server and create:
     // - Client attestation identity containing client's ephemeral public key
     // - Encryptor used for decrypting/encrypting messages between client and server
-    ClientIdentity clientIdentity = attestationEngine.processServerIdentity(serverIdentity);
+    byte[] clientIdentity = attestationEngine.processServerIdentity(serverIdentity);
     AttestedInvokeRequest clientIdentityRequest =
-        AttestedInvokeRequest.newBuilder().setClientIdentity(clientIdentity).build();
+        AttestedInvokeRequest.newBuilder().setAttestationMessage(
+            AttestationMessage.newBuilder().setBody(ByteString.copyFrom(clientIdentity)).build()
+        ).build();
```

Comment:
possibly for the future, but I don't even think this part of the logic should know about the various messages; it should ask the attestationEngine if there is any data to send, if so, send it (as bytes), and when the server replies, send that data to the attestation engine (as bytes). This class should not know anything about any inner message, they should only exist in the attestationEngine. 

The only output of attestationEngine is whether it expects more data from the other side (in which case it keeps reading it, in a loop), or whether it has succesfully derived the encryptor.

If we change the protocol to require an additional round trip, ideally this class should not change at all, right? If you think it should change instead, could you clarify why?